src/black/__init__.py

```diff
@@ -3306,10 +3316,26 @@ def do_transform(self, line: Line, string_idx: int) -> Iterator[TResult[Line]]:
                 yield TErr(
                     "Will not strip parentheses which have comments attached to them."
                 )
+                return
 
         new_line = line.clone()
         new_line.comments = line.comments.copy()
-        append_leaves(new_line, line, LL[: string_idx - 1])
+        try:
+            append_leaves(new_line, line, LL[: string_idx - 1])
+        except BracketMatchError as e:
+            # HACK: I believe there is currently a bug somewhere in
```

Comment:
This is unfortunate, but it's better than a crash, and the formatting in the newly added test cases looks reasonable.

Reply:
@JelleZijlstra On second thought, I don't think this will do. The `append_leaves()` function modifies the Node structure and so must be atomic. I can't fix this right now, but I will have to come up with a different solution.

Reply:
Ok. This is fixed. The corrected version of this fix, though still not ideal, has the benefit of stripping the parens as is consistent with blacks style after #1182.